antlr-kotlin-runtime/src/commonMain/kotlin/org/antlr/v4/kotlinruntime/misc/OrderedHashSet.kt

```diff
@@ -6,11 +6,83 @@
 
 package org.antlr.v4.kotlinruntime.misc
 
-/** A HashMap that remembers the order that the elements were added.
+/**
+ * A HashMap that remembers the order that the elements were added.
  * You can alter the ith element with set(i,value) too :)  Unique list.
- * I need the replace/set-element-i functionality so I'm subclassing
+ * I need the replace/set-element-i functionality, so I'm subclassing
  * LinkedHashSet.
  */
-class OrderedHashSet<T> : Set<T> by mutableSetOf() {
+public open class OrderedHashSet<T> private constructor(
+  private val delegate: LinkedHashSet<T>,
+) : MutableSet<T> by delegate {
+  public constructor() : this(LinkedHashSet())
 
+  /**
+   * Track the elements as they are added to the set.
+   *
+   * Note(Edoardo): renamed from "elements" to "myElements" to avoid a JS name clash
+   */
+  protected var myElements: ArrayList<T> = ArrayList()
```

Comment:
Pick a better name, or rename it just for JS.

Reply:
I believe there is a `@JsName` annotation that will rename it for js only.

Reply:
@frett yup, that's probably what I'm going to do.

Reply:
Done.